patches/0001-test-use-uk-test.h-in-test-api.c.patch

```diff
@@ -0,0 +1,468 @@
++
++	UK_TEST_EXPECT(p1 != NULL);
++	UK_TEST_EXPECT((uintptr_t)(p1) % 32 == 0);
++
++	void *p2 = mi_malloc_aligned(48, 32);
++
++	UK_TEST_EXPECT(p2 != NULL);
++	UK_TEST_EXPECT((uintptr_t)(p2) % 32 == 0);
++	mi_free(p2);
++	mi_free(p1);
++}
++
++UK_TESTCASE(libmimalloc_api_testsuite, malloc_aligned4)
++{
++	void *p;
++	bool ok = true;
++
++	for (int i = 0; i < 8 && ok; i++) {
++		p = mi_malloc_aligned(8, 16);
++		ok = (p != NULL && (uintptr_t)(p) % 16 == 0); mi_free(p);
++	}
++	UK_TEST_EXPECT(ok);
++}
++
++UK_TESTCASE(libmimalloc_api_testsuite, malloc_aligned_at1)
++{
++	void *p = mi_malloc_aligned_at(48, 32, 0);
++
++	UK_TEST_EXPECT(p != NULL);
++	UK_TEST_EXPECT(((uintptr_t)(p) + 0) % 32 == 0);
++	mi_free(p);
++}
++
++UK_TESTCASE(libmimalloc_api_testsuite, malloc_aligned_at2)
++{
++	void *p = mi_malloc_aligned_at(50, 32, 8);
++
++	UK_TEST_EXPECT(p != NULL);
++	UK_TEST_EXPECT(((uintptr_t)(p) + 8) % 32 == 0);
++	mi_free(p);
++}
++
++UK_TESTCASE(libmimalloc_api_testsuite, memalign1)
++{
++	void *p;
++	bool ok = true;
++
++	for (int i = 0; i < 8 && ok; i++) {
++		p = mi_memalign(16, 8);
++		ok = (p != NULL && (uintptr_t)(p) % 16 == 0); mi_free(p);
++	}
++	UK_TEST_EXPECT(ok);
++}
++
++// ---------------------------------------------------
++// Heaps
++// ---------------------------------------------------
++UK_TESTCASE(libmimalloc_api_testsuite, heap_destroy)
++{
++	mi_heap_t *heap = mi_heap_new();
++	int *p1 = mi_heap_malloc_tp(heap, int);
++	int *p2 = mi_heap_malloc_tp(heap, int);
++	*p1 = *p2 = 43;
++	mi_heap_destroy(heap);
++
++	UK_TEST_EXPECT(SHOULD_NOT_CRASH);
++}
++
++UK_TESTCASE(libmimalloc_api_testsuite, heap_delete)
++{
++	mi_heap_t *heap = mi_heap_new();
++	int *p1 = mi_heap_malloc_tp(heap, int);
++	int *p2 = mi_heap_malloc_tp(heap, int);
++
++	mi_heap_delete(heap);
++	*p1 = 42;
++	mi_free(p1);
++	mi_free(p2);
++
++	UK_TEST_EXPECT(SHOULD_NOT_CRASH);
++}
++
++// ---------------------------------------------------
++// various
++// ---------------------------------------------------
++UK_TESTCASE(libmimalloc_api_testsuite, realpath)
++{
++	void *s = &s;
++
++	s = (char *)mi_realpath(".", NULL);
++	UK_TEST_EXPECT(s != &s);
++	mi_free(s);
++	UK_TEST_EXPECT(SHOULD_NOT_CRASH);
++
++}
+ 
+-bool test_stl_allocator2() {
+ #ifdef __cplusplus
+-  std::vector<some_struct, mi_stl_allocator<some_struct> > vec;
+-  vec.push_back(some_struct());
+-  vec.pop_back();
+-  return vec.size() == 0;
+-#else
+-  return true;
++UK_TESTCASE(libmimalloc_api_testsuite, stl_allocator1)
++{
++	std::vector<int, mi_stl_allocator<int>> vec;
++	vec.push_back(1);
++	vec.pop_back();
++	UK_TEST_EXPECT_ZERO(vec.size());
++}
+ #endif
++
++#ifdef __cplusplus
++struct some_struct  { int i; int j; double z; };
++
++UK_TESTCASE(libmimalloc_api_testsuite, stl_allocator2)
++{
++	std::vector<some_struct, mi_stl_allocator<some_struct>> vec;
++	vec.push_back(some_struct());
++	vec.pop_back();
++	UK_TEST_EXPECT_ZERO(vec.size());
+ }
++#endif
++
++#if defined(CONFIG_LIBMIMALLOC_TEST_API)
```

Comment:
Since the last changes in the `Makefile.uk`, this is not needed anymore, the file being build only when `CONFIG_LIBMIMALLOC_TEST_API` is selected.
I would change this to make sure that `mimalloc` is selected as the default allocator in `ukboot`, to avoid the segfault.
Would look something like:
```c
  #ifdef CONFIG_LIBUKBOOT_INITMIMALLOC                                                                                                                                                                             
  uk_testsuite_register(libmimalloc_api_testsuite, NULL);                                                                                                                                                          
  #else                                                                                                                                                                                                            
  #error You need to select mimalloc as the default memory allocator \                                                                                                                                             
            inside the ukboot library for tests to run.                                                                                                                                                              
  #endif 
  ```

Reply:
Updated as per suggestions.